Sources/kai/Checker.swift

```diff
@@ -188,11 +188,14 @@ class Type: Equatable, CustomStringConvertible {
     }
 
     var isBooleanesque: Bool {
-        return !flags.union(.booleanesque).isEmpty
+        //FIXME(vdka): this is the old implementation and it ALWAYS returned true
+        //return !flags.union(.booleanesque).isEmpty
+        return flags.contains(.booleanesque) || isNumeric || isBoolean
```

Comment:
That being said, for clarities sake, it would 100% be neater to do `return isNumeric || isBoolean` so let's do that.